cli/cmd/cluster-apply.go

```diff
@@ -84,6 +84,13 @@ func runClusterApply(cmd *cobra.Command, args []string) {
 		ctxLogger.Fatalf("Verify cluster: %v", err)
 	}
 
+	labels := map[string]string{
+		"lokomotive.kinvolk.io/name": "kube-system",
+	}
+
+	if err := k8sutil.UpdateNamespaceWithLabels(labels, "kube-system", kubeconfigPath); err != nil {
```

Comment:
It seems to me that this is a great task for Kubernetes controller, to ensure that this label is always applied.

Reply:
@iaguis @surajssd @johananl

Reply:
Given that this namespace can't be deleted and users are not expected to remove labels from it I think it's fine to not have a controller. If a user changes labels they're deviating from Lokomotive so I'm not too worried about it.

Having said that, I think there's a compelling case for having a controller (or several) to handle this kind of tasks. This would be a bigger undertaking that would require design and considering what tasks we move there. I think it's out of scope for this PR.

Reply:
IMO this is not about namespaces being deleted or users removing labels, but having a declarative and clear approach, rather than imperative like here. Also, the controller could be even written in bash, similarly to what we've done with hostendpoint controller.

Reply:
I don't think this is imperative, we're using the `Update` API operation on a Kubernetes resource, we're just aligning that every time we do `cluster apply` manually (maybe that's what you mean by imperative?) instead of very often automatically. In this sense this is the same we do with components, we only align when the user does component/cluster apply.

Having a controller made sense for hostendpoints because they come and go, but here the `kube-system` NS is one, always there, and we don't expect users to change it.

I understand in this case you think a controller would be cleaner than this and I'm not against that, but I'd like to hear some compelling arguments on why we should do that before doing it.